File: Contest/Biweekly/Biweekly Contest 170/3752. Lexicographically Smallest Negated Permutation that Sums to Target/Solution.py
from typing import List

class Solution:
    def lexSmallestNegatedPerm(self, n: int, target: int) -> List[int]:
        # N : value of n
        # Time Complexity: O(N logN) -> for sorting
        # Space Complexity: O(N) -> with output array
        summ = (n * (n + 1)) // 2
        if target > summ or target < (summ * -1) or (summ + target) % 2 != 0:
            return []

        sumN = (summ - target) // 2

        res = []
        for num in range(n, 0, -1):
            if num <= sumN:
                sumN -= num
                res.append(-1 * num)
            else:
                res.append(num)
        res.sort()
        return res

File: Contest/Biweekly/Biweekly Contest 170/3752. Lexicographically Smallest Negated Permutation that Sums to Target/Solution.java
import java.util.*;

class Solution {
    public int[] lexSmallestNegatedPerm(int n, long target) {
        // N : value of n
        // Time Complexity: O(N logN) -> for sorting
        // Space Complexity: O(N) -> with output array
        long summ = ((long)n * (n + 1)) / 2;
        if (summ < target || (summ * -1) > target || (target + summ) % 2 != 0) return new int[]{};

        long sumN = (summ - target) / 2;
        int[] res = new int[n];
        int k = 0;
        for (int num = n; num > 0; num--) {
            if (num <= sumN) {
                sumN -= num;
                res[k++] = (num * -1);
            } else res[k++] = num;
        }
        Arrays.sort(res);
        return res;
    }
}